tests/roles/dataplane_adoption/tasks/main.yaml

```
@@ -1,5 +1,19 @@
+- name: use ansible-runner image built from source
+  when: ansibleee_runner_img is defined
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    {{ oc_login_command }}
+    # openstack-operator catalog pins the sha256, which might differ from latest with time
+    oc patch csv -n openstack-operators openstack-ansibleee-operator.v0.0.1 \
+      --type='json' -p='[{
+      "op":"replace", "path":"/spec/install/spec/deployments/0/spec/template/spec/containers/1/env/0",
+      "value": {"name": "RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT", "value": "{{ ansibleee_runner_img }}"}}]'
```

I think you can reuse the task below with something like this
```suggestion
      "value": {"name": "RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT", "value": "{{ ansibleee_runner_img|default('quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest') }}"}}]'
```

---

indeed, that's much cleaner, I'll change it in the next update, thanks @fao89